sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -337,6 +337,10 @@ trait CheckAnalysis extends PredicateHelper {
 
           case Tail(limitExpr, _) => checkLimitLikeClause("tail", limitExpr)
 
+          case Union(_, byName, allowMissingCol) if byName || allowMissingCol =>
+            failAnalysis("Union should not be with true `byName` or " +
+              "`allowMissingCol` flags after analysis phase.")
```

Comment:
Just a question; users can see this error message? That's the case of an analyzer bug?

Reply:
Usually not. This mainly prevents we accidentally create a Union with byName or allowMissingCol after ResolveUnion rule.

Reply:
And, yes, prevent a unexpected bug during analysis.